### PR TITLE
add example in Merge layer - Add()

### PR DESCRIPTION
**Pull request by keunwoochoi**
Because people seem have some problem understanding this new API, I added an example code.

https://github.com/fchollet/keras/issues/3921#issuecomment-308597601

### CODE REVIEW

**Comment by fchollet** [comment]
The syntax below would be preferred:

```python
s = layers.add([x1, x2])
```

**Comment by fchollet** [comment]
But we can show both and explain that they are equivalent

**Comment by keunwoochoi** [comment]
> The syntax below would be preferred:

Oh, really? But why? Because the example is functional? I thought `Add()([x1, x2])` should be preferred if not only allowed because then its API is consistent with others -- in the same form of `out = some_layer()(input)`

Another but related question: it is even possible to use any of Merge layer in a Sequential model? Because some people asked about it and also otherwise, showing `Add()` and `add()` could be a bit confusing I'm afraid. (+Those functional interfaces resemble backend functions..)

**Comment by keunwoochoi** [comment]
Maybe for the example of Add()
```python
sum = Add()([x1, x2])
# sum = add([x1, x2])
```
and the other way around for add(). But I guess there's a reason you keep both of them, so better examples would contain the most suitable cases.

File: keras/layers/merge.py [hunk 1]
```diff
@@ -187,6 +187,25 @@ class Add(_Merge):
...
     all of the same shape, and returns
     a single tensor (also of the same shape).
+
+    # Examples
+
+    ```python
+        from keras.layers.merge import Add  # a layer
+        from keras.layers.merge import add  # a functional interface
+        from keras.layers import Input, Dense
+        from keras.models import Model
```
Comment (fchollet) [inline keras/layers/merge.py]: Don't put import in example. Instead, prefer e.g. `keras.layers.Dense`

File: keras/layers/merge.py [hunk 2]
```diff
@@ -187,6 +187,25 @@ class Add(_Merge):
...
+
+        input1 = Input(shape=(16,))
+        x1 = Dense(8, activation='relu')(input1)
+        input2 = Input(shape=(32,))
+        x2 = Dense(8, activation='relu')(input2)
+        added = Add()([x1, x2])
+        # added = add([x1, x2])  # which is equivalent
```
Comment (fchollet) [inline keras/layers/merge.py]: "equivalent to: `keras.layers.add(...`"

File: keras/layers/merge.py [hunk 3]
```diff
@@ -187,6 +187,25 @@ class Add(_Merge):
...
+
+        out = Dense(4)(added)
+        model = Model(inputs=[input1, input2], outputs=out)
```
Comment (fchollet) [inline keras/layers/merge.py]: No need for this line

File: keras/layers/merge.py (hunk 1 above)
Comment (keunwoochoi) [inline keras/layers/merge.py]: For `Dense`, yes, but it seems necessary to make it clear that we can use
```python
import keras.layers.Add
from keras.layers.merge import Add
from keras.layers import Add
```
and if there's one that should be preferred, it should be mentioned somewhere (and perhaps why). I personally am still confused about it.

File: keras/layers/merge.py (hunk 2 above)
Comment (keunwoochoi) [inline keras/layers/merge.py]: Yeah.. but is the current one easier to understand? Well they are similar but the full line can be a bit easier for beginners I think?

File: keras/layers/merge.py (hunk 1 above)
Comment (fchollet) [inline keras/layers/merge.py]: > from keras.layers.merge import Add

That should never be used, and never be mentioned in docs/examples. The public API is `layers`.

Comment (keunwoochoi) [inline keras/layers/merge.py]: Hm.. okay, but is it mentioned in anyplace? I actually tried just all the stuff I can think of. My opinion is, for this purpose, examples would be better with import lines. What do you think? I'm pretty sure people have been importing layers in very different ways including some undersired ways cuz they don't know.

Comment (fchollet) [inline keras/layers/merge.py]: You can do:
`from keras import layers`
`layers.add`

Or:
`keras.layers.add`

File: keras/layers/merge.py [hunk 4]
```diff
@@ -187,6 +187,24 @@ class Add(_Merge):
...
     all of the same shape, and returns
     a single tensor (also of the same shape).
+
+
+
+    # Examples
+
+    ```python
+        import keras
+        from keras.models import Model
```
Comment (fchollet) [inline keras/layers/merge.py]: You could remove this import and just use `keras.models.Model`

File: keras/layers/merge.py [hunk 5]
```diff
@@ -451,6 +469,22 @@ def add(inputs, **kwargs):
 
     # Returns
         A tensor, the sum of the inputs.
+
+    # Examples
+
+    ```python
+        import keras
+        from keras.models import Model
```
Comment (fchollet) [inline keras/layers/merge.py]: Same

File: keras/layers/merge.py [hunk 6]
```diff
@@ -451,6 +468,21 @@ def add(inputs, **kwargs):
...
+    # Examples
+
+    ```python
+        import keras
+
+        input1 = keras.layers.Input(shape=(16,))
+        x1 = keras.layers.Dense(8, activation='relu')(input1)
+        input2 = keras.layers.Input(shape=(32,))
+        x2 = keras.layers.Dense(8, activation='relu')(input2)
+        added = keras.layers.add([x1, x2])  # which is equivalent
```
Comment (fchollet) [inline keras/layers/merge.py]: "# which is equivalent" -> did you mean to mention `Add`?

Comment (keunwoochoi) [inline keras/layers/merge.py]: Oh, mistake after copy-and-paste.

**Pull request merged**